### PR TITLE
Update to the new Prometheus library

### PR DESCRIPTION
**Pull request by addyess**

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -37,3 +37,8 @@ options:
     description: |
       Comma-separated list of Resources to override the default set of Resources to monitor.
     default: ""
+  scrape-interval:
```
Comment (addyess) [inline config.yaml:40]: this allows for flexible configuration of how often the endpoints are scraped by prometheus

File: src/charm.py [hunk 2]
```diff
@@ -91,14 +78,16 @@ def _validate_config(self):
...
-        binding = self.model.get_binding("monitoring")
+        binding = self.model.get_binding("metrics-endpoint")
         if not binding:
             return None
         return str(binding.network.ingress_address or binding.network.bind_address)
 
     @property
     def layer(self):
         """Pebble layer for workload."""
+        layer_args = list(self.config.keys())
```
Comment (addyess) [inline src/charm.py:89]: this excludes `scrape-interval` from showing up as a config to the metrics endpoint server

File: tests/integration/test_charm.py [hunk 3]
```diff
@@ -9,29 +9,17 @@
...
         },
+        trust=True,  # so that the container can access k8s api
     )
-    await ops_test.model.deploy(
-        prometheus_charm,
-        resources={
-            "prometheus-image": "prom/prometheus",
-        },
-    )
+    await ops_test.model.deploy("prometheus-k8s", channel="latest/beta")
```
Comment (addyess) [inline tests/integration/test_charm.py:22]: forego building the prometheus-k8s charm -- just deploy from charmhub

**Review by stonepreston — APPROVED** [review]
LGTM

**Pull request merged**